{
  "method": "sparse_with_keyphrases",
  "query": "TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.SpaceToBatchND` (in all backends such as XLA and handwritten kernels) is vulnerable to an integer overflow The result of this integer overflow is used to allocate the output tensor, hence we get a denial of service via a `CHECK`-failure (assertion failure), as in TFSA-2021-198. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.",
  "keyphrases": {
    "rootcause": "integer overflow"
  },
  "timestamp": "2025-03-18T12:32:45.407443",
  "results_count": 20,
  "results_summary": [
    {
      "doc_id": "CWE-1284",
      "name": "Improper Validation of Specified Quantity in Input",
      "score": 660.5430512086411
    },
    {
      "doc_id": "CWE-190",
      "name": "Integer Overflow or Wraparound",
      "score": 633.9292800906893
    },
    {
      "doc_id": "CWE-681",
      "name": "Incorrect Conversion between Numeric Types",
      "score": 628.472478658785
    },
    {
      "doc_id": "CWE-122",
      "name": "Heap-based Buffer Overflow",
      "score": 614.6946646723808
    },
    {
      "doc_id": "CWE-824",
      "name": "Access of Uninitialized Pointer",
      "score": 605.8656990334177
    }
  ]
}